sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -67,7 +67,7 @@ case class OptimizeSkewedJoin(conf: SQLConf) extends Rule[SparkPlan] {
    */
   private def isSkewed(size: Long, medianSize: Long): Boolean = {
     size > medianSize * conf.getConf(SQLConf.SKEW_JOIN_SKEWED_PARTITION_FACTOR) &&
-      size > conf.getConf(SQLConf.ADVISORY_PARTITION_SIZE_IN_BYTES)
+      size > conf.getConf(SQLConf.SKEW_JOIN_SKEWED_PARTITION_THRESHOLD)
```

Comment:
Do we still guarantee this after having s separate conf?
https://github.com/apache/spark/blob/30d95356f1881c32eb39e51525d2bcb331fcf867/sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala#L196

Reply:
now it's possible, if people set `SKEW_JOIN_SKEWED_PARTITION_THRESHOLD` much lower than `ADVISORY_PARTITION_SIZE_IN_BYTES`. But that's more like a malformed setting and we should simply ignore it.

Reply:
We'd better to update doc to warn user not to do so.